### PR TITLE
Remove multiple posts by altid

### PR DESCRIPTION
**Pull request by aamirafridi**
- Updated semistandard and fix all the **lint** issues
 - Add method to **remove** multiple posts by altid
 - Add method to **find** multiple posts by altid

### CODE REVIEW

File: api/post/index.js [hunk 1]
```diff
@@ -88,6 +85,17 @@ module.exports = function (api) {
     });
   }
 
+  function removePostsByAltid (keyspace, user, altid, next) {
+    getPostsByAltid(keyspace, user, altid, function (err, posts) {
+      if (err) { return next(err); }
+      async.map(posts, function (postItem, next2) {
```
Comment (tomruttle) [inline api/post/index.js]: use `cb` instead of `next2`?

File: api/post/index.js [hunk 2]
```diff
@@ -88,6 +85,17 @@ module.exports = function (api) {
...
+        _removePost(keyspace, postItem.post, next2);
+      }, function (err, status) {
```
Comment (tomruttle) [inline api/post/index.js]: Does this `err` not clash with the one above?

File: api/post/index.js [hunk 3]
```diff
@@ -124,14 +132,27 @@ module.exports = function (api) {
...
-        api.user.mapUserIdToUser(keyspace, post, ['user'], liu, next);
-      });
+      _validatePost(keyspace, liu, post, next);
+    });
+  }
+
+  function getPostsByAltid (keyspace, liu, altid, next) {
+    api.common.get(keyspace, 'selectPostByAltid', [altid], 'many', function (err, posts) {
+      if (err) { return next(err); }
+      async.map(posts, function (post, next2) {
```
Comment (tomruttle) [inline api/post/index.js]: use `cb` here instead of `next2`?

File: api/urls.js [hunk 4]
```diff
@@ -13,10 +13,12 @@ var urls = {
   addPost: '/post',
   getPost: '/post/:post',
   getPostByAltid: '/post/:altid/altid',
+  getPostsByAltid: '/post/:altid/altid',
```
Comment (tomruttle) [inline api/urls.js]: Is it okay that the above two urls are the same?

File: api/urls.js [hunk 5]
```diff
@@ -13,10 +13,12 @@ var urls = {
...
   updatePost: '/post/:post',
   updatePostByAltid: '/post/:altid/update/altid',
   removePost: '/post/:post',
   removePostByAltid: '/post/:altid/altid',
+  removePostsByAltid: '/post/:altid/altid',
```
Comment (tomruttle) [inline api/urls.js]: ^^ Ditto

File: tests/acceptance/api/posts.test.js [hunk 6]
```diff
@@ -195,7 +195,7 @@ databases.forEach(function (db) {
         });
       });
 
-      it('cant create, retrieve and delete a post by altid', function (done) {
+      it('can create, retrieve and delete a post by altid', function (done) {
```
Comment (tomruttle) [inline tests/acceptance/api/posts.test.js]: :laughing:

File: api/common/index.js [hunk 7]
```diff
@@ -20,8 +19,12 @@ module.exports = function (api) {
     return function (err, result) {
       /* istanbul ignore if */
       if (err) { return next(err); }
+      var noResultError = error(404, 'Item not found: "' + query + '"" for "' + data.join(', ') + '"');
       if (!result || (many !== 'many' && result.length !== 1)) {
-        return next(error(404, 'Item not found: "' + query + '"" for "' + data.join(', ') + '"'));
+        return next(error(404, noResultError));
+      }
+      if (many === 'many' && result && !result.length) {
```
Comment (tomruttle) [inline api/common/index.js]: You can probably merge these two `if` statements together?

Comment (tomruttle) [inline api/common/index.js]: Also, probably better using `result.length === 0` here as it's more explicit what you're doing.

File: api/post/index.js (hunk 1 above)
Comment (aamirafridi) [inline api/post/index.js]: Was trying to follow whats done here https://github.com/tes/seguir/blob/fbfd8db7b3f24560a98f67eb0394d023ddfe4f4c/api/feed/index.js#L92 `cb` and `cb2` but I think here `cb` makes sense. Changing this.

File: api/post/index.js (hunk 2 above)
Comment (aamirafridi) [inline api/post/index.js]: agree. es6 would show lint error for this. Changed second err to err2

Comment (aamirafridi) [inline api/post/index.js]: actually changed first to errGet and second to errRemove

File: api/post/index.js (hunk 3 above)
Comment (aamirafridi) [inline api/post/index.js]: done

File: api/urls.js (hunk 4 above)
Comment (aamirafridi) [inline api/urls.js]: Yes. https://github.com/tes/seguir/blob/remove-multiple-posts-by-altid/api/urls.js#L8-L9 and many more below these lines

File: api/urls.js (hunk 5 above)
Comment (aamirafridi) [inline api/urls.js]: its ok https://github.com/tes/seguir/pull/62#discussion_r50017300

File: tests/acceptance/api/posts.test.js (hunk 6 above)
Comment (aamirafridi) [inline tests/acceptance/api/posts.test.js]: 😺

**Comment by tomruttle** [comment]
:star: for you @aamirafridi. Will merge tomorrow morning.

**Comment by aamirafridi** [comment]
🐸

**Pull request merged**